.chloggen/fix-metric-mapping.yaml

```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: 'bug_fix'
+
+# The name of the component (e.g. pkg/quantile)
+component: pkg/otlp/metrics
+
+# A brief description of the change. Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Fix the incorrect for-loop which mutates the same slice while iterating over it, leading to unexpected metric names. Also avoid mutating the input pmetric.Metrics. 
```

Comment:
Does this mean that the Datadog exporter no longe mutates data? In that case we need to update it and change @mx-psi latest PR.

Reply:
Yeah we chatted about this, after collector and agent pick up this change, we can revert Pablo's PR.